modules/4337/contracts/Safe4337Module.sol

```diff
@@ -211,6 +211,41 @@ contract Safe4337Module is IAccount, HandlerContext, CompatibilityFallbackHandle
         operationHash = keccak256(operationData);
     }
 
+    /**
+     * @dev Checks if the signatures length is correct and does not contain additional bytes. The function does not check the integrity of the signature encoding.
+     * @param signatures signatures data
+     * @param threshold Indicates the number of iterations to perform in the loop.
+     * @return result True if length check passes, false otherwise.
+     */
+    function _checkSignatureLength(bytes calldata signatures, uint256 threshold) internal pure returns (bool result) {
+        uint256 offset = threshold * 0x41;
+
+        if (signatures.length < offset) return false;
+
+        result = true;
+        for (uint256 i = 0; i < threshold; i++) {
+            /* solhint-disable no-inline-assembly */
+            /// @solidity memory-safe-assembly
+            assembly {
+                let signaturePos := mul(0x41, i)
+                let s := calldataload(add(signatures.offset, add(signaturePos, 0x20)))
+                let v := byte(0, calldataload(add(signatures.offset, add(signaturePos, 0x40))))
```

Comment:
1. You can move the `s` to the if block, because it's only used there.
2. variable names can be more descriptive - since we're dealing with smart contract signatures, the values are not `s` and `v` per ser se, but `s` is a pointer to the start of the signature data and `v` is the signature type